substrate/frame/contracts/src/wasm/runtime.rs

```diff
@@ -558,6 +567,29 @@ impl<'a, E: Ext + 'a> Runtime<'a, E> {
 		self.ext.gas_meter_mut().adjust_gas(charged, token);
 	}
 
+	/// Charge, Run and adjust gas, for executing the given dispatchable.
+	fn call_dispatchable<F: FnOnce(&E) -> DispatchResultWithPostInfo>(
+		&mut self,
+		dispatch_info: DispatchInfo,
+		run: F,
+	) -> Result<ReturnCode, TrapReason> {
+		use frame_support::dispatch::extract_actual_weight;
+		let charged = self.charge_gas(RuntimeCosts::CallRuntime(dispatch_info.weight))?;
+		let result = run(self.ext);
+		let actual_weight = extract_actual_weight(&result, &dispatch_info);
+		self.adjust_gas(charged, RuntimeCosts::CallRuntime(actual_weight));
```

Comment:
using the sane RuntimeCosts::CallRuntime for call_runtime, and the new xcm_* calls below. Should we define one for each xcm call or using a generic one is good enough?

Reply:
My understanding is that they are duplicated so that we can print different label when debugging. Not too sure how useful it is in practice